flat_kivy_extensions/flatappextension.py

```diff
@@ -0,0 +1,366 @@
+                    Color:
+                        rgb: root.background_color
+                    Rectangle:
+                        pos: self.pos
+                        size: self.size
+
+                ScreenManager:
+                    id: screenmanager
+                    transition: NoTransition()
+                    # Question: When selecting FadeTransition the switching
+                    #           annimation seems to go black when switching,
+                    #           how can this be avoided?
+                    transition: FadeTransition()
+                    canvas.before:
+                        Color:
+                            rgb: root.background_color
+                        Rectangle:
+                            pos: self.pos
+                            size: self.size
+
+
+<-HeaderLayout>:
+    header_height: '40dp'
+    header_color: 0.2, 0.1, 0.2
+    menu_button_width: '50dp'
+    title: 'Header'
+    color: .1, .9, .1
+
+    canvas.before:
+        Color:
+            rgb: root.color
+        Rectangle:
+            pos: self.pos
+            size: self.size
+
+    CustomIconButton:
+        id: _menu_button
+        theme: ('app', 'header')
+        icon: 'fa-bars'
+        size_hint_x: None
+        width: root.menu_button_width
+        color: root.header_color
+
+    FlatLabel:
+        text: root.title
+        theme: ('app', 'header')
+
+''')
+
+
+Builder.load_file('flat_kivy_extensions/ui_elements.kv')
+
+class HeaderLayout(BoxLayout):
+    pass
+
+class RootWidget(Widget):
+    pass
+
+
+class ExtendedFlatApp(FlatApp):
+    def __init__(self, app_config_entries, title, about, *largs, **kwargs):
+        super(ExtendedFlatApp, self).__init__(*largs, **kwargs)
+        self.title = title
+        self.app_config_entries = app_config_entries
+        self.about = about
+        self.lazy_loading = False
+        self._first_screen = None
+        self._first_navigation_label = None
+
+    def build(self):
+        self.root = RootWidget()
+        # Question: Is this the best place to get references to
+        #           widgets with ids in the root widget?
+        self._navigationdrawer = self.root.ids.navigationdrawer
+        self._side_panel = self.root.ids.side_panel
+        self._header = self.root.ids.header
+        self._screenmanager = self.root.ids.screenmanager
+        self._menu_button = self._header.ids._menu_button
+        self._menu_button.bind(on_press=lambda j: self._navigationdrawer.toggle_state())
+
+
+        entry_constructors = {type(str()) : self._create_navigation_label_from_string,
+                              type(dict()) : self._create_navigation_label_from_dict,
+                              type(tuple()) : self._create_navigation_button,
+                              }
+
+        for entry in self.app_config_entries:
+            entry_constructors[type(entry)](entry)
+
+        return self.root
+
+    def setup_themes(self):
+        main = {
```

Comment:
I suggest moving the theme dict declarations to another py file and then importing them and loading them here